src/Symfony/Component/Process/Process.php

```diff
@@ -327,6 +322,11 @@ public function start(callable $callback = null, array $env = [])
             // last exit code is output on the fourth pipe and caught to work around --enable-sigchild
             $descriptors[3] = ['pipe', 'w'];
 
+            if (\is_array($commandline)) {
+                // exec is mandatory to deal with sending a signal to the process
+                $commandline = 'exec '.$this->buildShellCommandline($commandline);
+            }
```

Comment:
I’m not sure if `exec ` is needed here, on my Platform (macOS) it does not make a difference.

Reply:
I think so, let's keep it.